File: app/src/main/java/ai/doc/netrunner/view/ClassificationViewModel.java
package ai.doc.netrunner.view;

import androidx.lifecycle.ViewModel;

import ai.doc.netrunner.ModelRunner;
import ai.doc.tensorio.TIOModel.TIOModelBundleManager;

public class ClassificationViewModel extends ViewModel {
    private int currentTab = -1;
    private ModelRunner modelRunner;
    private TIOModelBundleManager manager;

    public ModelRunner getModelRunner() {
        return modelRunner;
    }

    public void setModelRunner(ModelRunner modelRunner) {
        this.modelRunner = modelRunner;
    }

    public TIOModelBundleManager getManager() {
        return manager;
    }

    public void setManager(TIOModelBundleManager manager) {
        this.manager = manager;
    }

    public int getCurrentTab() {
        return currentTab;
    }

    public void setCurrentTab(int currentTab) {
        this.currentTab = currentTab;
    }
}

File: app/src/main/java/ai/doc/netrunner/view/ClassificationViewModel.kt
package ai.doc.netrunner.view

import ai.doc.netrunner.ModelRunner
import ai.doc.tensorio.TIOModel.TIOModelBundleManager
import androidx.lifecycle.ViewModel
import java.util.ArrayList

class ClassificationViewModel : ViewModel() {
    enum class Tab {
        LiveVideo,
        TakePhoto,
        ChoosePhoto
    }

    lateinit var manager: TIOModelBundleManager
    lateinit var modelRunner: ModelRunner
    var currentTab = Tab.LiveVideo

    val modelIds: ArrayList<String> by lazy {
        ArrayList<String>(manager.bundleIds)
    }
}